retrieval_evaluator/retrieval_evaluator.py

```diff
@@ -0,0 +1,558 @@
+# retrieval_evaluator.py
+
+import json
+import shutil
+import matplotlib.pyplot as plt
+import numpy as np
+import os
+from pathlib import Path
+import pandas as pd
+from matplotlib.gridspec import GridSpec
+from matplotlib.patches import Patch
+from scipy import stats
+import time
+import logging
+from fuzzywuzzy import fuzz
+from collections import defaultdict
+from typing import List, Dict, Any, Union
+import argparse
+
+class RetrievalEvaluator:
+    def __init__(self):
+        self.project_root = Path(__file__).parent
+        self.dev_bench_reference_data_path = self.project_root / "dev_bench_reference_data"
+        self.comparison_data_path = self.project_root / "comparison_data"
+        self.logs_path = self.project_root / "logs"
+        self.evaluation_outputs_path = self.project_root / "evaluation_outputs"
+        self.logger = logging.getLogger("retrieval_evaluator")
+        self.results = []
+        self.summary = {}
+        self.enhanced_summary = {}
+
+        # Create necessary directories
+        self.project_root.mkdir(exist_ok=True)
+        self.dev_bench_reference_data_path.mkdir(exist_ok=True)
+        self.comparison_data_path.mkdir(exist_ok=True)
+        self.logs_path.mkdir(exist_ok=True)
+        self.evaluation_outputs_path.mkdir(exist_ok=True)
+
+    def setup_logging(self, benchmark_version: str):
+        log_file = self.logs_path / f'evaluation_{benchmark_version}.log'
+        self.logger.setLevel(logging.INFO)
+        formatter = logging.Formatter('%(asctime)s - %(levelname)s - %(message)s', datefmt='%Y-%m-%d %H:%M:%S')
+
+        # Remove any existing handlers to avoid duplicate logging
+        for handler in self.logger.handlers[:]:
+            self.logger.removeHandler(handler)
+
+        file_handler = logging.FileHandler(log_file)
+        file_handler.setLevel(logging.INFO)
+        file_handler.setFormatter(formatter)
+
+        self.logger.addHandler(file_handler)
+
+    def clear_comparison_data(self):
+        if self.comparison_data_path.exists():
+            os.remove(self.comparison_data_path)
+        self.logger.info(f"Cleared previous comparison data from {self.comparison_data_path}")
+
+    def load_json_file(self, file_path):
+        try:
+            with open(file_path, 'r') as file:
+                return json.load(file)
+        except FileNotFoundError:
+            self.logger.warning(f"File not found: {file_path}")
+            return []
+        except json.JSONDecodeError:
+            self.logger.error(f"Error decoding JSON from file: {file_path}")
+            return []
+
+    def is_trivia_memory(self, memory):
+        return "Here are some trivia questions and answers for you to process." in memory.get('query', '')
+
+    def fuzzy_match(self, query1, query2, threshold=85):
+        return fuzz.ratio(query1.lower(), query2.lower()) >= threshold
```

Comment:
You mentioned that this was giving you some issues. In other parts of the benchmark I've used the `standardise` and `rouge_l` functions from `utils.text` to accomplish a similar goal, but I believe that ROUGE is even more strict than L-distance.

Reply:
I will have a look into this